### PR TITLE
fix kicbase dockerfile to allow ppc64le and armv7 archs

### LINKED ISSUE #12399: Fix building for ppc64le and arm(v7) in kicbase
Building for ppc64le in kicbase was failing so we removed it (#12398) so we could release, ppc64le should be revisited and added back.

### PR DESCRIPTION
**Pull request by sharifelgamal**
Fixes #12399

### CODE REVIEW

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

**Comment by sharifelgamal** [comment]
ok-to-build-image

File: deploy/kicbase/Dockerfile [hunk 1]
```diff
@@ -159,19 +162,27 @@ RUN export ARCH=$(dpkg --print-architecture | sed 's/ppc64el/ppc64le/' | sed 's/
...
+RUN export ARCH=$(dpkg --print-architecture | sed 's/ppc64el/ppc64le/') && \
+    if [ "$ARCH" != "ppc64le" ]; then sh -c "echo 'deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \
+    	curl -LO https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_20.04/Release.key && \
+    	apt-key add - < Release.key && \
+    	clean-install containers-common catatonit conmon containernetworking-plugins cri-tools podman-plugins crun; \
+    else sh -c "echo 'deb https://download.opensuse.org/repositories/devel:/kubic:/libcontainers:/stable/xUbuntu_20.04/ /' > /etc/apt/sources.list.d/devel:kubic:libcontainers:stable.list" && \
+    	curl -LO https://download.opensuse.org/repositories/devel:kubic:libcontainers:stable/xUbuntu_20.04/Release.key && \
+    	apt-key add - < Release.key && \
+    	clean-install containers-common conmon containernetworking-plugins crun; \
+    fi
```
Comment (spowelljr) [inline deploy/kicbase/Dockerfile:174]: I believe just the `clean-install` line needs to be in the `if`, the rest seems identical

File: cmd/minikube/cmd/start.go [hunk 2]
```diff
@@ -1267,25 +1267,30 @@ func validateDiskSize(diskSize string) error {
...
 		// Convert `cri-o` to `crio` as the K8s config uses the `crio` spelling
-		if runtime == "cri-o" {
+		if rtime == "cri-o" {
 			viper.Set(containerRuntime, constants.CRIO)
 		}
+
+	}
+
+	if (rtime == "crio" || rtime == "cri-o") && (strings.HasPrefix(runtime.GOARCH, "ppc64") || detect.RuntimeArch() == "arm" || strings.HasPrefix(detect.RuntimeArch(), "arm/")) {
+		out.Error(reason.RuntimeIncompat, "The {{.runtime}} runtime is not compatible with the {{.arch}} architecture.", out.V{"runtime": rtime, "arch": runtime.GOARCH})
```
Comment (medyagh) [inline cmd/minikube/cmd/start.go:1289]: how about  we just exit here and link to github issue on CRIO github ?

**Review by medyagh — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/start.go (hunk 2 above)
Comment (sharifelgamal) [reply cmd/minikube/cmd/start.go:1289]: returning an error will effectively exit here, changed this to returning an error and added a link to the issue

File: deploy/kicbase/Dockerfile (hunk 1 above)
Comment (sharifelgamal) [reply deploy/kicbase/Dockerfile:174]: good call, fixed.

**Comment by sharifelgamal** [comment]
ok-to-build-image

File: cmd/minikube/cmd/start.go [hunk 3]
```diff
@@ -1267,25 +1267,30 @@ func validateDiskSize(diskSize string) error {
...
 
 		// Convert `cri-o` to `crio` as the K8s config uses the `crio` spelling
-		if runtime == "cri-o" {
+		if rtime == "cri-o" {
 			viper.Set(containerRuntime, constants.CRIO)
 		}
+
+	}
+
+	if (rtime == "crio" || rtime == "cri-o") && (strings.HasPrefix(runtime.GOARCH, "ppc64") || detect.RuntimeArch() == "arm" || strings.HasPrefix(detect.RuntimeArch(), "arm/")) {
```
Comment (spowelljr) [inline cmd/minikube/cmd/start.go:1288]:
```suggestion
	if (rtime == "crio" || rtime == "cri-o") && (strings.HasPrefix(runtime.GOARCH, "ppc64le") || detect.RuntimeArch() == "arm" || strings.HasPrefix(detect.RuntimeArch(), "arm/")) {
```
I'm assuming we want the `le` version?

**Review by spowelljr — CHANGES_REQUESTED** [review]

File: cmd/minikube/cmd/start.go (hunk 3 above)
Comment (spowelljr) [reply cmd/minikube/cmd/start.go:1288]: Has prefix, missed that

**Review by spowelljr — APPROVED** [review]

**Pull request merged**